MobiusCore/Source/EffectHandlers/EffectCallback.swift

```diff
@@ -54,17 +64,26 @@ public final class EffectCallback<Output> {
         }
     }
 
+    /// Send a number of events to the Mobius loop, then `end()` this callback.
+    /// Note: Calling this function after calling `.end()` is a no-op.
+    /// - Parameter outputs: The events which should be sent to the loop.
     public func end(with outputs: Output...) {
         end(with: outputs)
     }
 
+    /// Send a number of events to the Mobius loop, then `end()` this callback.
+    /// Note: Calling this function after calling `.end()` is a no-op.
+    /// - Parameter outputs: The events which should be sent to the loop.
     public func end(with outputs: [Output]) {
         for output in outputs {
```

Comment:
This actually means that someone else might call `end()` and interrupt the flow of outputs, or that this set of outputs might be mixed with another invocation of `send`. I think it would be better to set `ended` to true at the beginning of this method, then invoke the `onSend` closure instead of the `send` method.